src/process/linux.rs

```diff
@@ -0,0 +1,861 @@
+    /// exit status is not available at this time then `Ok(None)` is returned.
+    /// If an error occurs, then that error is returned.
+    ///
+    /// Note that unlike `wait`, this function will not attempt to drop stdin.
+    ///
+    /// # Examples
+    ///
+    /// Basic usage:
+    ///
+    /// ```no_run
+    /// use birdcage::process::Command;
+    /// use birdcage::{Birdcage, Sandbox};
+    ///
+    /// let cmd = Command::new("ls");
+    /// let mut child = Birdcage::new().spawn(cmd).unwrap();
+    ///
+    /// match child.try_wait() {
+    ///     Ok(Some(status)) => println!("exited with: {status}"),
+    ///     Ok(None) => {
+    ///         println!("status not ready yet, let's really wait");
+    ///         let res = child.wait();
+    ///         println!("result: {res:?}");
+    ///     },
+    ///     Err(e) => println!("error attempting to wait: {e}"),
+    /// }
+    /// ```
+    pub fn try_wait(&mut self) -> io::Result<Option<ExitStatus>> {
+        // Wait for child process to exit.
+        unsafe {
+            let mut status: libc::c_int = 0;
+            let pid = libc::waitpid(self.pid as i32, &mut status, libc::WNOHANG);
+            if pid == -1 {
+                Err(io::Error::last_os_error())
+            } else if pid == 0 {
+                Ok(None)
+            } else {
+                match self.exit_signal()? {
+                    Some(exit_signal) => Ok(Some(exit_signal)),
+                    None => Ok(Some(ExitStatus::from_raw(status))),
+                }
+            }
+        }
+    }
+
+    /// Simultaneously waits for the child to exit and collect all remaining
+    /// output on the stdout/stderr handles, returning an `Output`
+    /// instance.
+    ///
+    /// The stdin handle to the child process, if any, will be closed
+    /// before waiting. This helps avoid deadlock: it ensures that the
+    /// child does not block waiting for input from the parent, while
+    /// the parent waits for the child to exit.
+    ///
+    /// By default, stdin, stdout and stderr are inherited from the parent.
+    /// In order to capture the output into this `Result<Output>` it is
+    /// necessary to create new pipes between parent and child. Use
+    /// `stdout(Stdio::piped())` or `stderr(Stdio::piped())`, respectively.
+    ///
+    /// # Examples
+    ///
+    /// ```should_panic
+    /// use birdcage::process::{Command, Stdio};
+    /// use birdcage::{Birdcage, Sandbox};
+    ///
+    /// let mut cmd = Command::new("/bin/cat");
+    /// cmd.arg("file.txt");
+    /// cmd.stdout(Stdio::piped());
+    /// let child = Birdcage::new().spawn(cmd).expect("failed to execute child");
+    ///
+    /// let output = child.wait_with_output().expect("failed to wait on child");
+    ///
+    /// assert!(output.status.success());
+    /// ```
+    pub fn wait_with_output(mut self) -> io::Result<Output> {
+        // Explicitly drop stdin, to avoid deadlocks.
+        drop(self.stdin.take());
```

Comment:
Either use `.take()` or use `drop()`, but not both...

```suggestion
        // Explicitly drop stdin, to avoid deadlocks.
        let _ = self.stdin.take();
```

```suggestion
        // Explicitly drop stdin, to avoid deadlocks.
        drop(self.stdin);
```

Reply:
The second thing you've used doesn't work, which is why it's the way it is. I can use `let _` instead but I used an explicit drop since the semantics can be a bit weird between `let _` and `let _a`.